scalyr_agent/configuration.py

```diff
@@ -359,23 +362,39 @@ def parse(self):
                     "serverHost"
                 ] = self.__get_default_hostname()
 
-            # Add in implicit entry to collect the log generated by this agent.
+            # Add in implicit entry to collect the logs generated by this agent and its worker sessions.
             agent_log = None
+            worker_session_agent_logs = None
             if self.implicit_agent_log_collection:
                 # set path as glob to handle log files from the multiprocess worker sessions.
-                config = JsonObject(
-                    path="agent*.log",
-                    exclude=JsonArray("*_debug.log"),
+                config = JsonObject(path="agent.log", parser="scalyrAgentLog",)
+
+                # add log config for the worker session agent log files.
+                worker_session_logs_config = JsonObject(
+                    path="%s*.log" % AGENT_WORKER_SESSION_LOG_NAME_PREFIX,
+                    # exclude debug files for the worker session debug logs.
+                    # NOTE: the glob for the excluded files has to be restrictive enough,
+                    # so it matches only worker session debug logs.
+                    exclude=JsonArray(
+                        "*%s*_debug.log" % AGENT_WORKER_SESSION_LOG_NAME_PREFIX
```

Comment:
IIRC, if we want actual debug logs to be ingested, user needs to add a manual entry to the config, right (aka we don't have a specific config flag for it, right)?

Reply:
Right. A new log matcher should be created so there won't be those excluded paths. If i am not mistaken, we are already doing that for the `agent_debug.log` file in smoke tests.

Reply:
Btw, do you think it would be better to introduce this config option?

Reply:
A config option to turn on collection of debug logs?  I'm not sure.  Debug logs can add a fair bit of noise and the users get charged for it.  So, we haven't made it easy for them to accidentally turn them on.  I think we can hold with that for now.

Reply:
Yeah, fair point, agreed with Steven, not needed at this point.